imap_processing/ultra/l0/decom_tools.py

```diff
@@ -182,9 +182,9 @@ def decompress_image(
     blocks_per_row = cols // pixels_per_block
 
     # Compressed pixel matrix
-    p = np.zeros((rows, cols), dtype=np.int16)
+    p = np.zeros((rows, cols), dtype=np.uint16)
```

Comment:
Reading the comments below I think that `p` should be `np.uint8` because you want to make this value into the range 0-255.
```suggestion
    p = np.zeros((rows, cols), dtype=np.uint8)
```